runtime_scan/pkg/provider/aws/config.go

```
@@ -15,40 +15,89 @@
-	SecurityGroupID string // the scanner's security group
-	InstanceType    string // the scanner's instance type
+	// Region where the Scanner instance needs to be created
+	ScannerRegion string
+	// SubnetID where the Scanner instance needs to be created
+	SubnetID string
+	// SecurityGroupID which needs to be attached to the Scanner instance
+	SecurityGroupID string
+	// KeyPairName is the name of the SSH KeyPair to use for Scanner instance launch
+	KeyPairName string
+	// ScannerImage is the AMI image used for creating Scanner instance
+	ScannerImage string
+	// ScannerInstanceType is the instance type used for Scanner instance
+	ScannerInstanceType string
+	// BlockDeviceName contains the block device name used for attaching Scanner volume to the Scanner instance
+	BlockDeviceName string
 }
 
-func setConfigDefaults() {
-	viper.SetDefault(AWSJobImageID, defaultAWSJobImageID)
-	viper.SetDefault(AWSInstanceType, defaultAWSInstanceType)
+func (c *Config) Validate() error {
+	if c.ScannerRegion == "" {
+		return fmt.Errorf("parameter Region must not be nil")
+	}
+
+	if c.SubnetID == "" {
+		return fmt.Errorf("parameter SubnetID must not be nil")
+	}
+
+	if c.SecurityGroupID == "" {
+		return fmt.Errorf("parameter SecurityGroupID must not be nil")
+	}
+
+	if c.ScannerImage == "" {
+		return fmt.Errorf("parameter ScannerImage must not be nil")
+	}
+
+	if c.ScannerInstanceType == "" {
+		return fmt.Errorf("parameter ScannerInstanceType must not be nil")
+	}
 
-	viper.AutomaticEnv()
+	return nil
 }
 
-func LoadConfig() *Config {
-	setConfigDefaults()
+func NewConfig() (*Config, error) {
+	// Avoid modifying the global instance
+	v := viper.New()
+
+	v.AllowEmptyEnv(true)
+	v.AutomaticEnv()
+
+	_ = v.BindEnv("ScannerRegion", EnvScannerRegion)
+	_ = v.BindEnv("SubnetID", EnvSubnetID)
+	_ = v.BindEnv("SecurityGroupID", EnvSecurityGroupID)
+	_ = v.BindEnv("KeyPairName", EnvKeyPairName)
+	_ = v.BindEnv("ScannerImage", EnvScannerImage)
```

not sure we want to be eating the errors here right? I'm not sure what the errors might be but it would probably be a good idea to handle them

---

We are eating errors as `BindEnv` always returns a `nil` in this case.